ludwig/data/split.py

```diff
@@ -152,31 +151,36 @@ def get_schema_cls():
 
 
 def stratify_split_dataframe(
-    df: DataFrame, column: str, probabilities: List[float], random_seed: float
+    df: DataFrame, column: str, probabilities: List[float], backend: Backend, random_seed: float
 ) -> Tuple[DataFrame, DataFrame, DataFrame]:
     """Splits a dataframe into train, validation, and test sets based on the values of a column.
 
     The column must be categorical (including binary). The split is stratified, meaning that the proportion of each
     category in each split is the same as in the original dataset.
     """
+
     frac_train, frac_val, frac_test = probabilities
 
-    # Dataframe of just the column on which to stratify
-    y = df[[column]]
-    try:
-        df_train, df_temp, _, y_temp = train_test_split(
-            df, y, stratify=y, test_size=(1.0 - frac_train), random_state=random_seed
-        )
-        # Split the temp dataframe into val and test dataframes.
-        relative_frac_test = frac_test / (frac_val + frac_test)
-        df_val, df_test, _, _ = train_test_split(
-            df_temp, y_temp, stratify=y_temp, test_size=relative_frac_test, random_state=random_seed
-        )
-    except ValueError:
-        raise ValueError(
-            "Stratified splitting cannot be performed because the least populated class has too few examples. "
-            "Consider adding more data, consolidating rare classes by setting vocab_size, or using a random split."
-        )
+    def _safe_stratify(df, column, test_size):
```

Comment:
Let's add a test to verify this new functionality.

Reply:
Added